File: Code/Algorithms/linearsearch.py
#Code to impement linear search

def ls(arr,l,x):
    for i in range(0,l):
        if(arr[i]==x):
            return i
    return -1


if __name__=="__main__":
    arr=[]
    y=int(input("Enter the number of elements:"))
    if(y!=0):
        print("Enter the numbers:")
    for a in range(0,y):
        b=int(input())
        arr.append(b)


    x=int(input("Enter the number to be searched:"))
    l=len(arr)

    z=ls(arr,l,x)
    if(z==-1):
        print("Number not found in the array")
    else:
        print("Number found at the position", z)

File: Code/Algorithms/linearsearch.c
//Code to implement linear search using c

#include <stdio.h>
 
int search(int arr[], int N, int x)
{
    int i;
    for (i = 0; i < N; i++)
        if (arr[i] == x)
            return i;
    return -1;
}
 

int main(void)
{
    int y;
    printf("Enter the no. of elements: ");
    scanf("%d",&y);
    int arr[y];
    
    int i;
    
    if(y!=0)
        printf("Enter elements of the array - \n");

    for(i=0;i<y;i++)
    {
        scanf("%d",&arr[i]);
    }
    
    
    int x;
    printf("Enter the no. to be searched in th array: ");
    scanf("%d",&x);
    int N = sizeof(arr) / sizeof(arr[0]);
 
 
 
    int result = search(arr, N, x);
    
    if (result == -1)
        printf("Element is not present in array");
        
    else 
        printf("Element is present at position %d", result+1);
        
    return 0;
}
